src/corelibs/U2Algorithm/src/misc/RollingMatrix.h

```diff
@@ -32,7 +34,6 @@ class U2ALGORITHM_EXPORT RollingMatrix {
 public:
     RollingMatrix(int _sizeX, int _sizeY)
         : sizeX(_sizeX), sizeY(_sizeY), column0(0) {
-        // TODO: check memory overflow on the client side!
         U2_ASSERT(sizeX >= 0 && sizeY >= 0);
```

Comment:
Why not SAFE_POINT as we discussed?

Reply:
Because you wrote "This should be assertion because it creates a matrix in invalid state.", if you mean c++ assert - it is useless in release mode, so i decided what you talking about U2_ASSERT. I still can't understand you want here.

Reply:
SAFE_POINT is the way  UGENE' asserts in release mode and restores from internal errors. 
Please use SAFE_POINTS and never user U2_ASSERT/assertion.

If this is the last use of U2_ASSERT you can remove it from UGENE codebase